### DISCUSSION TITLE
BitOps V2.0.0 - Post release tweaks and suggestions

### DISCUSSION THREAD

**Discussion by PhillypHenning** [bitovi/bitops#249] [General]
Please post up any thoughts on tweaks or improvements to core functionality.

Use threads to break up discussions

**Comment by ConnorGraham** [comment]
### Improved Error Messaging
 it would be good to give increased help/error messaging to bitops in a future release. In general an error message could:
1. Provide a guess at what the error is
2. Provide a link to documentation

I'm going to create a distinct thread for each example

#### Example 1 No BITOPS_ENVIRONMENT var
```
$ docker run bitovi/bitops:latest
...
2022-08-03 16:22:39,280 root         ERROR ENVIRONMENT variables must be set... Exiting
```

#### Example 1 Better Output
> Missing BITOPS_ENVIRONMENT environment variable (`-e BITOPS_ENVIRONMENT=/path/to/ops/repo`) for more information see link-to-docs

**Reply by PhillypHenning** [reply]
Updated to 
`The `BITOPS_ENVIRONMENT` variable must be set... Exiting\nFor more information on this issue please checkout our doc [https://bitovi.github.io/bitops/configuration-base/#environment]"`

**Comment by ConnorGraham** [comment]
### Improved Error Messaging
 it would be good to give increased help/error messaging to bitops in a future release. In general an error message could:
1. Provide a guess at what the error is
2. Provide a link to documentation

I'm going to create a distinct thread for each example

#### Example 2 BITOPS_ENVIRONMENT , but no volume mount
```
$ docker run -e BITOPS_ENVIRONMENT=hello bitovi/bitops:latest
...
Traceback (most recent call last):
  File "/opt/bitops/scripts/plugins.py", line 47, in <module>
    Deploy_Plugins()
  File "/opt/bitops/scripts/plugins/deploy_plugins.py", line 60, in Deploy_Plugins
    copy_tree(bitops_deployment_dir, temp_dir)
  File "/usr/local/lib/python3.8/site-packages/setuptools/_distutils/dir_util.py", line 139, in copy_tree
    raise DistutilsFileError("cannot copy tree '%s': not a directory" % src)
distutils.errors.DistutilsFileError: cannot copy tree '/opt/bitops_deployment/': not a directory
```

#### Example 2 Better Output
> Cannot find operations repo, ensure you are mounting your operations repo as a volume (`-v $(pwd)/opt/bitops_deployment`) and your operations repo contains a ${BITOPS_ENVIRONMENT} (literally use the BITOPS_ENVIRONMNET value here!) directory. For more information on operations repo's see link-to-docs

**Reply by PhillypHenning** [reply]
Logging was added, check whether operations repo was mounted and if not displays the following message and exits.
`"An operations repo needs to be mounted to the Docker container with the path `/opt/bitops_deployment/`... Exiting.\n\t\tFor more information on this issue please checkout our doc [https://bitovi.github.io/bitops/about/#how-bitops-works]"`

**Comment by mickmcgrath13** [comment]
also see: https://github.com/bitovi/bitops/issues/247

**Reply by PhillypHenning** [reply]
Addressed all of these concerns 👍
